src/libmeasurement_kit/dns/system_resolver_impl.hpp

```diff
@@ -1,194 +0,0 @@
-        assert(p->ai_family == AF_INET or p->ai_family == AF_INET6);
-        if (p->ai_family == AF_INET) {
-            answer.type = MK_DNS_TYPE_A;
-            addr_ptr = &((sockaddr_in *)p->ai_addr)->sin_addr;
-        } else if (p->ai_family == AF_INET6) {
-            answer.type = MK_DNS_TYPE_AAAA;
-            addr_ptr = &((sockaddr_in6 *)p->ai_addr)->sin6_addr;
-        } else {
-            // Added branch to avoid g++ compiler warning
-            throw std::runtime_error("internal error");
-        }
-        if (p->ai_canonname != nullptr) {
-            answer.hostname = p->ai_canonname;
-        }
-        if (inet_ntop(p->ai_family, addr_ptr, abuf, sizeof(abuf)) == nullptr) {
-            ctx->logger->warn("dns: unexpected inet_ntop failure");
-            ctx->reactor->call_soon(
-                [=]() { callback(InetNtopFailureError(), nullptr); });
-            return;
-        }
-        if (p->ai_family == AF_INET) {
-            answer.ipv4 = std::string(abuf);
-        } else if (p->ai_family == AF_INET6) {
-            answer.ipv6 = std::string(abuf);
-        }
-        message->answers.push_back(answer);
-    }
-    ctx->reactor->call_soon([=]() { callback(NoError(), message); });
-}
-
-template <MK_MOCK(getaddrinfo), MK_MOCK(inet_ntop)>
-void system_resolver_impl(QueryClass dns_class, QueryType dns_type,
-        std::string name, Callback<Error, Var<Message>> cb, Settings settings,
-        Var<Reactor> reactor, Var<Logger> logger) {
-    std::unique_ptr<ResolverContext> ctx(new ResolverContext(
-        dns_class, dns_type, name, cb, settings, reactor, logger));
-    Query query;
-    /*
-     * Note: here we pass empty flags. It used to be AI_ALL | AI_V4MAPPED but
-     * Android did not like it and, honestly, I do not think having back v4
-     * mapped addresses would be super useful to us (or in general) given that
-     * when we want to connect we usually connect trying to resolve AAAA and
-     * A at the same time. Keeping the feature conditionally may lead to
-     * odd behaviors in corner cases and I'd like to avoid future headaches.
-     */
-    ctx->hints.ai_socktype = SOCK_STREAM;
-
-    if (dns_class != MK_DNS_CLASS_IN) {
-        reactor->call_soon([=]() { cb(UnsupportedClassError(), nullptr); });
-        return;
-    }
-
-    if (dns_type == MK_DNS_TYPE_A) {
-        ctx->hints.ai_family = AF_INET;
-    } else if (dns_type == MK_DNS_TYPE_AAAA) {
-        ctx->hints.ai_family = AF_INET6;
-    } else if (dns_type == MK_DNS_TYPE_CNAME) {
-        ctx->hints.ai_family = AF_UNSPEC;
-        ctx->hints.ai_flags |= AI_CANONNAME;
-    } else {
-        reactor->call_soon([=]() { cb(UnsupportedTypeError(), nullptr); });
-        return;
-    }
-
-    query.type = dns_type;
-    query.qclass = dns_class;
-    query.name = name;
-
-    ctx->message->queries.push_back(query);
-
-    std::async(std::launch::async, resolve_async_impl<getaddrinfo, inet_ntop>,
-               ctx.release());
```

Comment:
This is the piece of code move in `system_resolver.hpp` and modified as explained above.